buildSrc/src/main/java/com/google/firebase/gradle/plugins/FirebaseJavaLibraryPlugin.kt

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.firebase.gradle.plugins
+
+import com.github.sherter.googlejavaformatgradleplugin.GoogleJavaFormatExtension
+import com.google.common.collect.ImmutableList
+import com.google.common.collect.ImmutableMap
+import java.io.File
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.api.attributes.Attribute
+import org.gradle.api.plugins.JavaPluginConvention
+import org.gradle.api.publish.PublishingExtension
+import org.gradle.api.publish.maven.MavenPublication
+import org.gradle.kotlin.dsl.getPlugin
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+class FirebaseJavaLibraryPlugin : Plugin<Project> {
+
+  override fun apply(project: Project) {
+    project.apply(ImmutableMap.of("plugin", "java-library"))
+    project.apply(ImmutableMap.of("plugin", "com.github.sherter.google-java-format"))
+    project.extensions.getByType(GoogleJavaFormatExtension::class.java).toolVersion = "1.10.0"
+    val firebaseLibrary =
+      project.extensions.create(
+        "firebaseLibrary",
+        FirebaseLibraryExtension::class.java,
```

Comment:
With `org.gradle.kotlin.dsl.create` you can reduce this to:
```kotlin
val firebaseLibrary =
      project.extensions.create<FirebaseLibraryExtension>(
        "firebaseLibrary",
        project,
        LibraryType.ANDROID
      )
```

And if you add an import for `com.google.firebase.gradle.plugins.LibraryType.ANDROID`, you could also reduce it to two lines:
```kotlin
val firebaseLibrary =
      project.extensions.create<FirebaseLibraryExtension>("firebaseLibrary", project, ANDROID)
```

But that's your choice :)